Initial commit advisory opinions landing page

porta-antiporta opened the pull request.

Commit:
initial commit

porta-antiporta commented:
i don't understand why code coverage is going UP with this merge.  if anything it should go DOWN.

porta-antiporta commented:
There's still some outstanding work (populating the `Pending` and `Latest` advisory opinions)  but appreciate a review on what is currently in place.  Tried following existing patterns.

adborden commented on openfecwebapp/routes.py:
```diff
@@ -253,12 +254,12 @@ def legal_search(query, result_type):
 def advisory_opinions(query, offset):
     result_type = 'advisory_opinions'
     results = {}
-
```
Watch these whitspace changes, they do have meaning to separate logical code blocks.

adborden commented on openfecwebapp/routes.py:
```diff
@@ -253,12 +254,12 @@ def legal_search(query, result_type):
-    return views.render_legal_doc_search_results(results, query, result_type)
+        return views.render_legal_doc_search_results(results, query, result_type)
+    else: 
+        return views.render_legal_advisory_opinion_landing()
```
I would put this in it's own route. I don't feel strongly about the endpoint names, though. Maybe the existing `advisory-opinions` and `regulations` search should move under `/legal/search/advisory-opinions` and `/legal/search/regulations`. which would free up `/legal/advisory-opinions/` for the landing page.

Eventually, I was envisioning something like `/legal/advisory-opintions/<AO number>` for the canonical AO pages, so doing something like the above fits with that.

adborden commented on openfecwebapp/templates/legal_advisory_opinion_landing.html:
```diff
@@ -0,0 +1,50 @@
+{% import 'macros/legal-search.html' as legal_search %}
+{% import 'macros/breadcrumbs.html' as breadcrumb %}
+
+{% set breadcrumb_links=[(cms_url + '/legal-resources', 'Legal resources')] %}
```
This should include another level of bread crumb for "Advisory opinions".

adborden replied:
Oh, nevermind, this is covered when we call the macro.

adborden commented on openfecwebapp/templates/legal_advisory_opinion_landing.html:
```diff
@@ -0,0 +1,50 @@
+  {% else %}
+  Search {{ document_type_display_name }}
+  {% endif %}
+{% endblock %}
```
This isn't necessary anymore, could just be "Advisory opinions".

adborden commented on openfecwebapp/templates/legal_advisory_opinion_landing.html:
```diff
@@ -0,0 +1,50 @@
+
+{% block body %}
+<header class="page-header slab slab--primary">
+  {{ breadcrumb.breadcrumbs(document_type_display_name, breadcrumb_links) }}
```
We can hardcode `document_type_display_name` with "Advisory opinions" since it's never going to be anything else anymore.

adborden commented on openfecwebapp/templates/legal_advisory_opinion_landing.html:
```diff
@@ -0,0 +1,50 @@
+    <div class="results-info results-info--simple">
+    <h2 class="results-info__title" id="advisory-opinions">Latest advisory opinions
+    </h2>
+    </div>
```
The indenting here looks off.

adborden commented on openfecwebapp/templates/legal_advisory_opinion_landing.html:
```diff
@@ -0,0 +1,50 @@
+      <p class="u-padding--top">Advisory opinions are official Commission responses to questions about how federal campaign finance law applies to specific, factual situations.
+      </p>
+      <p class="u-padding--top">This archive contains advisory opinions from 1975 to the present. It also contains documents related to advisory opinions — such as requests, drafts and public comments — from 1990 to the present.
+      </p>
```
Can you clean up the tags here, the open tag starts inline, but the close tag is on it's own line. I'd prefer it be consistent, both on their own line or both inline.

adborden commented on openfecwebapp/templates/legal_advisory_opinion_landing.html:
```diff
@@ -0,0 +1,50 @@
+      <p class="usa-width-one-half">This site is in beta, which means we're testing content. This content is not finalized, is not legal advice and is subject to change.</p>
+      <p class="usa-width-one-half">Everything on this site should be read in conjunction with FEC.gov. Please let us know what you think of our new content; use the feedback tool on this page.</p>
+    </div>
+</div>
```
We should pull this into a partial template. I suspect we should be including it on other legal pages as well, no?

adborden commented:
> i don't understand why code coverage is going UP with this merge. if anything it should go DOWN.
 @porta-antiporta

Since the template is loaded in the tests, adding lines to the template will increase code coverage.

adborden commented:
> There's still some outstanding work (populating the Pending and Latest advisory opinions) but appreciate a review on what is currently in place. Tried following existing patterns.

:+1: did you want to tackle that in this PR, or a new one?

adborden committed:
Rename search routes

adborden committed:
Clean up AO landing page

adborden committed:
Move legal disclaimer into partial

adborden committed:
Refactor legal filter into macro

Allows the search filter to exist on a page different than the search pages
(like the landing page).

adborden committed:
Clean up search pages

- Remove scrollbar on results table
- Lowercase document type names

adborden committed:
Disable pending/latest AO sections

adborden committed:
Fix whitespace

adborden committed:
Merge remote-tracking branch 'origin/develop' into ao_landing_page

Github can't resolve the merge itself.

adborden commented:
cc @anthonygarvan @jenniferthibault @emileighoutlaw

adborden commented:
![screenshot from 2016-08-10 17-37-41](https://cloud.githubusercontent.com/assets/509703/17575781/323e5248-5f21-11e6-9a5b-06ccd9957ec0.png)

adborden committed:
Use the correct layout styles

adborden commented:
@jenniferthibault pointed out we weren't using the correct layout styles to get the sections formatted properly. Hopefully I did it right but @noahmanger can you take a look?

@jenniferthibault can you double check this looks correct?
![screenshot from 2016-08-11 18-55-11](https://cloud.githubusercontent.com/assets/509703/17610699/793b3d62-5ff5-11e6-883a-0c95472970b0.png)

noahmanger commented on openfecwebapp/templates/legal-advisory-opinion-landing.html:
```diff
@@ -0,0 +1,53 @@
+<section class="main__content--full data-container__wrapper">
+  {{ legal.filter('Search advisory opinions', 'advisory_opinions') }}
+  <div class="content__section data-container__body">
+    <div class="main">
```
@jenniferthibault do we want to use `.main` here? It adds padding above and below the div so it doesn't line up with the filters title:
![image](https://cloud.githubusercontent.com/assets/1696495/17613877/75abf54e-6016-11e6-8bcb-509902ea9129.png)

noahmanger commented on openfecwebapp/templates/legal-advisory-opinion-landing.html:
```diff
@@ -0,0 +1,53 @@
+
+<section class="main__content--full data-container__wrapper">
+  {{ legal.filter('Search advisory opinions', 'advisory_opinions') }}
+  <div class="content__section data-container__body">
```
I just noticed that in my refactor of the filter panel / data-container layout, I removed the left padding on `.data-container__body`, which works for when it's actually a data table, but means that in this layout, the content runs up to the edge of the screen.

noahmanger replied:
And on mobile there's no padding: 
![image](https://cloud.githubusercontent.com/assets/1696495/17614068/0485e080-6018-11e6-8e59-ca7702ed7085.png)

noahmanger commented on openfecwebapp/templates/macros/legal.html:
```diff
@@ -1,3 +1,28 @@
+{% macro filter(title, result_type, query) %}
+<div id="filters" class="filters filters--fixed is-open">
+  <div class="filters__hider">
```
You don't need `.filters__hider` (which was renamed `.filters__content` in https://github.com/18F/fec-style/pull/446/ ) because these are always open. That div is just for hiding them.

noahmanger commented on openfecwebapp/templates/macros/legal.html:
```diff
@@ -1,3 +1,28 @@
+{% macro filter(title, result_type, query) %}
+<div id="filters" class="filters filters--fixed is-open">
+  <div class="filters__hider">
+    <h2 class="filter__header filters__inner">
```
I removed the `.filter__header` class, so you don't need it here.

noahmanger commented on openfecwebapp/templates/macros/legal.html:
```diff
@@ -1,3 +1,28 @@
+        <div class="filter" id="search-field">
+          <label class="label" for="search-input">Search terms</label>
+          <div class="combo combo--search--mini">
+            <ul class="dropdown__selected"></ul>
```
You can remove this `ul`

adborden replied:
derp.

noahmanger commented:
By and large looks good. Made a few tiny notes in places, catching things where the refactor of the filter panel had minor implications.

adborden commented:
Great, I'll clean up the filter classes. Did you want to tackle the [data table padding](https://github.com/18F/openFEC-web-app/pull/1394#discussion_r74544318)?

noahmanger commented:
Yeah I can get to that later today before publishing to npm.

adborden committed:
Remove unused filter classes

emileighoutlaw commented:
Content looks good to me. Thanks @porta-antiporta and @adborden

adborden committed:
Remove right margin on legal filters

adborden committed:
Merge branch 'develop' into ao_landing_page

adborden committed:
Fix test for renamed url
